tests/test_algorithm.py

```diff
@@ -627,50 +640,66 @@ def test_order_methods_for_future(self, algo_class):
+            method_name=order_style,
+            env=self.env
+        )
+        algo.run(self.data_portal)
+
+    @parameterized.expand([
+        (TestOrderAlgorithm,),
+        (TestOrderValueAlgorithm,),
+        (TestTargetAlgorithm,),
+        (TestOrderPercentAlgorithm,),
+        (TestTargetPercentAlgorithm,),
+    ])
+    def test_minute_data(self, algo_class):
+        tempdir = TempDirectory()
 
-        method_names_to_test = ['order',
-                                'order_value',
-                                'order_percent',
-                                'order_target',
-                                'order_target_percent',
-                                'order_target_value']
+        try:
+            env = TradingEnvironment()
 
-        for name in method_names_to_test:
-            algo = TestOrderStyleForwardingAlgorithm(
-                sim_params=self.sim_params,
-                instant_fill=False,
-                method_name=name,
-                env=self.env
+            sim_params = SimulationParameters(
+                period_start=pd.Timestamp('2002-1-2', tz='UTC'),
+                period_end=pd.Timestamp('2002-1-4', tz='UTC'),
+                capital_base=float("1.0e5"),
+                data_frequency='minute',
+                env=env
             )
-            algo.run(self.data_portal)
 
-    def test_order_instant(self):
-        algo = TestOrderInstantAlgorithm(sim_params=self.sim_params,
-                                         env=self.env,
-                                         instant_fill=True)
-        algo.run(self.data_portal)
+            equities_metadata = {}
 
-    def test_minute_data(self):
-        sim_params = SimulationParameters(
-            period_start=pd.Timestamp('2002-1-2', tz='UTC'),
-            period_end=pd.Timestamp('2002-1-4', tz='UTC'),
-            capital_base=float("1.0e5"),
-            data_frequency='minute',
-            env=self.env
-        )
+            for sid in [0, 1]:
+                equities_metadata[sid] = {
+                    'start_date': sim_params.period_start,
+                    'end_date': sim_params.period_end
+                }
 
-        self.data_portal = create_data_portal(
-            self.env,
-            self.tempdir,
-            sim_params,
-            [0, 1]
-        )
+            env.write_data(equities_data=equities_metadata)
 
-        algo = TestOrderInstantAlgorithm(sim_params=sim_params,
-                                         env=self.env,
-                                         instant_fill=True)
-        algo.run(self.data_portal)
+            data_portal = create_data_portal(
+                env,
+                tempdir,
+                sim_params,
+                [0, 1]
+            )
+
+            algo = algo_class(sim_params=sim_params,
+                                      env=env)
+            algo.run(data_portal)
+        finally:
+            tempdir.cleanup()
```

Comment:
this is a tempdir created specifically in this test, since we had to create a new dataportal in this test.